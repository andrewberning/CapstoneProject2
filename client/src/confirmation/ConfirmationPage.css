.ConfirmationPage {
  max-width: 600px;
  margin: 0 auto;
}

.confirmation-container {
  background-color: #fff;
  border-radius: 8px;
}

.confirmation-details {
  font-size: 1.1rem;
}