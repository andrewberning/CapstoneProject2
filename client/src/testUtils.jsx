import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
