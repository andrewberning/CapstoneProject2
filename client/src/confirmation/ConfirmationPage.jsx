import { Link } from 'react-router-dom';
import "./ConfirmationPage.css";

export default function ConfirmationPage() {
  return (
    <div className="ConfirmationPage container">
      <h2 className="text-center my-4">Confirmation Page</h2>
      <div className="confirmation-container card p-4 shadow-sm">
        <div className="confirmation-details">
          <p className="mb-4">Your order has been received!</p>
          <p className="mb-4">Thanks for Shopping with us at Shoply!</p>
        </div>
        <div className="btn-container">
          <Link to={"/"} className="btn btn-primary">Home</Link>
        </div>
      </div>
    </div>
  )
}