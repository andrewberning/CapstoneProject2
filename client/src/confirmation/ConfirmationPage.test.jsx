import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ConfirmationPage from "./ConfirmationPage"
import { UserProvider } from "../testUtils";

describe('ConfirmationPage component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <ConfirmationPage />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <ConfirmationPage />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
})