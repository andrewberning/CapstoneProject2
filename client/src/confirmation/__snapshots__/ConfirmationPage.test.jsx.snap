// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfirmationPage component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ConfirmationPage container"
  >
    <h2
      class="text-center my-4"
    >
      Confirmation Page
    </h2>
    <div
      class="confirmation-container card p-4 shadow-sm"
    >
      <div
        class="confirmation-details"
      >
        <p
          class="mb-4"
        >
          Your order has been received!
        </p>
        <p
          class="mb-4"
        >
          Thanks for Shopping with us at Shoply!
        </p>
      </div>
      <div
        class="btn-container"
      >
        <a
          class="btn btn-primary"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
